### ISSUE TITLE
Augmentation fails after reaching to the end of stream

### ISSUE THREAD

**Issue by salmans** [salmans/Razor#72]
Assume that a theory has two models. After performing `next` twice, Razor says that there is no more models for the given theory. Here, our contract with the user is that the current model is set to the last model of the theory. However, if the user augments the "current model", Razor says that the augmentation is inconsistent.

**Comment by ryandanas** [comment]
Question, does it work just fine if the interaction is...
(at first model) ```next``` (at second model) ```augment```
... instead of...
(at first model) ```next``` (at second model) ```next``` (no more models; at second still) ```augment```

**Comment by salmans** [comment]
yes

**Comment by ryandanas** [comment]
then i believe fixing this issue will be as easy as setting the SMT solver iterator back once if it fails to find a model. i hope to fix this before our 3pm meeting.

**Comment by salmans** [comment]
Great! I Thanks

**Comment by dandougherty** [comment]
By the way, this reminds me of something I noticed earlier.

> Assume that a theory has two models. After performing next twice, Razor
> says that there is no more models for the given theory
>
In general, when Razor gets to the end of the sequence of models in the min
set of support, does it (still) say "no more models!" or something like
that?     That's almost never literally true, right?
A better response would be "no more minimal models" or, more positively,
"I've reached the end of the minimal models.   To see more models, try
augmentation."
Or something like that.
